captum/_utils/gradient.py

```diff
@@ -23,7 +23,9 @@
 )
 
 
-def apply_gradient_requirements(inputs: Tuple[Tensor, ...]) -> List[bool]:
+def apply_gradient_requirements(
+    inputs: Tuple[Tensor, ...], warn: bool = False
```

Comment:
nit: In order to support original behavior don't we want to not warn only when we know that warning is not necessary in case of layer approaches ?

Reply:
Great catch, thanks! Yes, you're definitely right, meant to set the default to True